### PR TITLE
CLN: fix mypy error pandas/tests/plotting/test_backend.py

### PR DESCRIPTION
**Pull request by saska**
GH28926, mypy didn't like setting non-existant attribute on module.
Dummy function, circumvented with `__setattr__`

- [x] addresses part of #28926
- [x] tests added / passed
- [x] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/plotting/test_backend.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import pandas
 
 dummy_backend = types.ModuleType("pandas_dummy_backend")
-dummy_backend.plot = lambda *args, **kwargs: None
+dummy_backend.__setattr__("plot", lambda *args, **kwargs: None)
```
Comment (simonjayhawkins) [inline pandas/tests/plotting/test_backend.py]: use setattr() (or a # type: ignore) see https://github.com/python/mypy/issues/7171#issuecomment-520407743

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@saskakarsi Thanks for the PR. minor change, also can you merge master

File: pandas/tests/plotting/test_backend.py (hunk 1 above)
Comment (saskakarsi) [reply pandas/tests/plotting/test_backend.py]: Done! Personally preferred the setattr() way over ignoring a test line here, comment didn't specify why monkey-patching would be bad here - can do it the other way as well if you prefer and we just want to skip the line entirely.

Comment (simonjayhawkins) [reply pandas/tests/plotting/test_backend.py]: Thanks for the update, but can you review the merge. looks like something went awry.

Comment (saskakarsi) [reply pandas/tests/plotting/test_backend.py]: Oh jeez. Will do!

Comment (saskakarsi) [reply pandas/tests/plotting/test_backend.py]: Alright, Travis CI wanted to run a couple of times but merge should be fine now!

**Review by simonjayhawkins — APPROVED** [review]
@saskakarsi lgtm @WillAyd

**Comment by WillAyd** [comment]
@TomAugspurger mind taking a look at this one?

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**